### PR TITLE
Rename image-builder default branch to main

### PR DESCRIPTION
**Pull request by AverageMarcus**
The image-builder project is planning to rename the default branch from `master` to `main`. This PR is in preparation for that happening as outlined in https://www.kubernetes.dev/resources/rename/.

Towards: https://github.com/kubernetes-sigs/image-builder/issues/1161

/assign @mboersma @kkeshavamurthy @jsturtevant
/hold

### CODE REVIEW

**Comment by jsturtevant** [comment]
is it possible to reference both master and main?  that would make transition easier.

**Comment by AverageMarcus** [comment]
Not as far as I was able to tell. There is an open issue about updating the tooling to allow just using the default branch but that isn’t in place yet. https://github.com/kubernetes/test-infra/issues/20672

**Comment by AverageMarcus** [comment]
@mboersma We're still going to need approval from others as this PR is updating other projects jobs that make use of image-builder in their pipelines. Mainly just CAPI providers so we can reach out to them on Slack to get this approved (and remain on hold) before we do the rename.

**Comment by richardcase** [comment]
For CAPA & CAPG:

/approve

**Comment by AverageMarcus** [comment]
@richardcase it doesn't look like you're listed in the OWNERS file for CAPG in test-infra 😞  - https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cluster-api-provider-gcp/OWNERS

**Review by cpanato — APPROVED** [review]
/approve
/lgtm

**Comment by AverageMarcus** [comment]
Thanks everyone! Really appreciate your fast response! 

This will now remain on hold until after we've completed the rename.

**Comment by AverageMarcus** [comment]
I had to rebase with some merged changes that altered the indentation. 😞  

Would someone mind dropping a LGTM please? 🙏

**Comment by AverageMarcus** [comment]
/unhold

Branch has been renamed

**Pull request merged**
